nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AddPermission.java

```diff
@@ -110,8 +114,8 @@ protected void addPermission(DocumentModel doc) {
         }
 
         String username = user;
-        if (username == null) {
-            username = NuxeoPrincipal.computeTransientUsername(email);
+        if (!isUserFound(username)) {
```

Comment:
I don't agree that we should create a transient username when `user == null && email != null`, because this is the objective of this ticket: deny creating permission if user is not found. Instead, maybe we can do

```java
if (!isUserNameFound(user) || !isUserEmailFound(email)) { ... }
```

Would it be a good idea?